### PR TITLE
feat(metanode): optimize mp leader send snapshot to follower

### PR DESCRIPTION
**Pull request by true1064**
**What this PR does / why we need it**:
(1) add a configuration item for raft retainLogs, for metaNode; 
(2) add a configuration item to set which format version to use when mp raft leader send snapshot to follower; 
(3) when follower apply snapshot, won't fail if format snapshot version not match, just handle the part of fields it can recognize;

(2 )and (3) are the  focus of this revision，to handle the compatibility of snapshot format of meta partitifion when upgrading. 

**Special notes for your reviewer**:

### CODE REVIEW

File: metanode/metanode.go [hunk 1]
```diff
@@ -248,6 +250,34 @@ func (m *MetaNode) parseConfig(cfg *config.Config) (err error) {
...
+			log.LogInfof("invalid config raftSyncSnapFormatVersion, using default[%v]", m.raftSyncSnapFormatVersion)
+		} else {
+			m.raftSyncSnapFormatVersion = raftSyncSnapFormatVersion
+			log.LogInfof("by config raftSyncSnapFormatVersion:[%v]", m.raftSyncSnapFormatVersion)
+		}
+	} else {
+		m.raftSyncSnapFormatVersion = SnapFormatVersion_1
+		log.LogInfof("using default raftSyncSnapFormatVersion[%v]", m.raftSyncSnapFormatVersion)
+	}
+	syslog.Println("conf raftSyncSnapFormatVersion=", m.raftSyncSnapFormatVersion)
```
Comment (xiangcai1215) [inline metanode/metanode.go:278]: 这个信息需要打印出来吗，还是只有日志就好了

File: util/config/config.go [hunk 2]
```diff
@@ -157,7 +157,12 @@ func (c *Config) GetInt64(key string) int64 {
 	return 0
 }
 
-// GetBool returns a int64 value for the config key.
+func (c *Config) HasKey(key string) bool {
+	_, present := c.data[key]
```
Comment (xiangcai1215) [inline util/config/config.go:161]: present直接用ok呗，golang更常见一点

**Review by xiangcai1215 — APPROVED** [review]
两个评论看看要不要改，流程没问题

**Review by Victor1319 — APPROVED** [review]
LGTM

File: util/config/config.go (hunk 2 above)
Comment (true1064) [reply util/config/config.go:161]: util/config/config.go这个文件里用的都是present，不改了

**Pull request merged**